src/pages/index.vue

```diff
@@ -154,61 +142,8 @@ const HomePage = {
       })),
     }))
 
-    const featuredSearchIdx = ref(-1) // immediately updates to 0 on mount
-    const nextIdx = () => {
-      featuredSearchIdx.value =
-        (featuredSearchIdx.value + 1) % featuredSearches.length
-      const featuredSearch = featuredSearches[featuredSearchIdx.value]
-      typeText(featuredSearch.term, () => {
-        setImages(featuredSearch.images)
-      })
-    }
-
-    const rotationTime = 6e3 // ms
-    let rotInterval = null
-    onMounted(() => {
-      nextIdx()
-      rotInterval = setInterval(nextIdx, rotationTime)
-      document.addEventListener('visibilitychange', () => {
-        if (document.visibilityState === 'visible') {
-          rotInterval = setInterval(nextIdx, rotationTime)
-        } else if (document.visibilityState === 'hidden' && rotInterval) {
-          clearInterval(rotInterval)
-        }
-      })
-    })
-    onBeforeUnmount(() => {
-      if (rotInterval) {
-        clearInterval(rotInterval)
-      }
-    })
-
-    const featuredSearchText = ref('')
-    const keystrokeTime = 100 // ms
-    const typeText = (text, callback) => {
-      let length = 0
-      const typeInterval = setInterval(() => {
-        length += 1
-        featuredSearchText.value = text.substring(0, length)
-        if (length === text.length) {
-          clearInterval(typeInterval)
-          callback()
-        }
-      }, keystrokeTime)
-    }
-
-    const featuredSearchImages = ref([])
-    const setImages = (images) => {
-      featuredSearchImages.value = images
-    }
-
-    const isMounted = ref(false)
-    onMounted(() => {
-      isMounted.value = true
-    })
-    onBeforeUnmount(() => {
-      isMounted.value = false
-    })
+    const featuredSearchIdx = Math.floor((Math.random() * 10) % 3)
```

Comment:
Nit: Since `Math.random()` is `[0, 1)`, multiplying by 3 gives `[0, 3)`, which can be floored to give `{0, 1, 2}` directly.
```suggestion
    const featuredSearchIdx = Math.floor(Math.random() * 3)
```